lambda/printlambda.ml

```diff
@@ -536,85 +553,127 @@ let rec lam ppf = function
           id_arg_list in
       fprintf ppf
         "@[<2>(letrec@ (@[<hv 1>%a@])@ %a)@]" bindings id_arg_list lam body
-  | Lprim(prim, largs, _) ->
+  | Lprim(prim, largs, loc) ->
       let lams ppf largs =
         List.iter (fun l -> fprintf ppf "@ %a" lam l) largs in
-      fprintf ppf "@[<2>(%a%a)@]" primitive prim lams largs
-  | Lswitch(larg, sw, _loc) ->
+      fprintf ppf "@[<2>(%a%a%a)@]"
+        primitive prim
+        print_loc loc
+        lams largs
+  | Lswitch(larg, sw, loc) ->
       let switch ppf sw =
         let spc = ref false in
         List.iter
-         (fun (n, l) ->
+         (fun (n,act) ->
            if !spc then fprintf ppf "@ " else spc := true;
-           fprintf ppf "@[<hv 1>case int %i:@ %a@]" n lam l)
+           fprintf ppf "@[<hv 1>case int %i:%a@ %a@]"
+             n
+             print_loc act.block_loc
+             lam act.expr)
```

Comment:
you should have a `block` or `print_block` function that does the right thing so that this can be rewritten as

```ocaml
fprintf ppf "[@<hv 1>case int %i:%a@]" n block act
```